include/git2/diff.h

```diff
@@ -414,10 +414,10 @@ typedef struct {
 
 	/**
 	 * A size (in bytes) above which a blob will be marked as binary
-	 * automatically; pass a negative value to disable.
+	 * automatically; pass `GIT_OBJECT_SIZE_MAX` to disable.
 	 * Defaults to 512MB.
 	 */
-	git_off_t   max_size;
+	git_object_size_t max_size;
```

Comment:
Hm, this changes our API, doesn't it?

Reply:
In practice, if callers ever provided a negative value, it was caught in `diff_file_content_init_common`, and ignored.  
 https://github.com/libgit2/libgit2/pull/5123#discussion-diff-317085663R49 was doing the same.  So as-is we are API compatible, but if we remove that odd-looking check, then we are in fact changing the API.

Reply:
Moved this back to a signed int for sanity.